federation/pkg/kubefed/init/init.go

```diff
@@ -240,6 +241,20 @@ func initFederation(cmdOut io.Writer, config util.AdminConfig, cmd *cobra.Comman
 		if err != nil {
 			return err
 		}
+	}
+
+	//as the federation control plane is up and responding
+	//we create a config map in the fed control plane
+	//which subsequently is consumed by the dns servers of the yet to be registered clusters
+	//expecting that when ever a new cluster registers with it
+	//the control plane will create a config map in that cluster
```

Comment:
This is very cool! I hadn't thought about it this way at all! Thanks for dogfooding!

Have you tested this on a real setup though? I am wondering if all this works without creating `kube-system` namespace in federation.